shared/src/api/extension/api/configuration.test.ts

```diff
@@ -2,66 +2,51 @@ import { initNewExtensionAPI } from '../flatExtensionApi'
 import { SettingsEdit } from '../../client/services/settings'
 import { pretendRemote } from '../../util'
 import { MainThreadAPI } from '../../contract'
+import { SettingsCascade } from '../../../settings/settings'
+
+const initialSettings = (value: { a: string }): SettingsCascade<{ a: string }> => ({
+    subjects: [],
+    final: value,
+})
 
 describe('ConfigurationService', () => {
     describe('get()', () => {
-        test("throws if initial settings haven't been received", () => {
-            const { configuration } = initNewExtensionAPI(pretendRemote({}))
-            expect(() => configuration.get()).toThrow('unexpected internal error: settings data is not yet available')
-        })
         test('returns the latest settings', () => {
             const {
                 configuration,
                 exposedToMain: { syncSettingsData },
-            } = initNewExtensionAPI(pretendRemote({}))
+            } = initNewExtensionAPI(pretendRemote({}), initialSettings({ a: 'a' }))
             syncSettingsData({ subjects: [], final: { a: 'b' } })
             syncSettingsData({ subjects: [], final: { a: 'c' } })
             expect(configuration.get<{ a: string }>().get('a')).toBe('c')
         })
     })
 
     describe('changes', () => {
-        test('emits as soon as initial settings are received', () => {
-            const {
-                configuration,
-                exposedToMain: { syncSettingsData },
-            } = initNewExtensionAPI(pretendRemote({}))
+        test('emits immediately on subscription with initial settings (why btw?)', () => {
```

Comment:
Because it is more convenient for extensions (otherwise almost every extension would have to  add `startWith({})`). I think the new test title is not quite correct though, because it doesn't emit the settings, it emits `void` afaik
```suggestion
        test('emits immediately on subscription', () => {
```